Adding a note about Fusion and Snowflake key pairs

runleonarun opened the pull request:
## What are you changing in this pull request and why?

Adding more opinion to Snowflake key pair auth on fusion. From internal [Slack convo](https://dbt-labs.slack.com/archives/C08AA468V54/p1758922494978519?thread_ts=1757449188.763049&cid=C08AA468V54).

## Checklist
<!-- Ensure your PR meets the following items. Feel free to add additiona checklist items to the list if additional checks need to happen before the PR is merged, such as:
- [ ] Needs technical review
- [ ] Change base branch
- [ ] Merge on xyz date
-->
- [ ] The changes in this PR meet the [docs style guide/fundamentals](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) required for all content.
- [ ] Applied the proper versioning rules if the content is for specific dbt version(s): ([version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).).
<!--
PRE-RELEASE VERSION OF dbt (if so, uncomment):
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
<!-- 
ADDING OR REMOVING PAGES (if so, uncomment):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages
-->

<!-- vercel-deployment-preview -->
---
🚀 Deployment available! Here are the direct links to the updated files:


- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/cloud/billing
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/cloud/dbt-cloud-ide/keyboard-shortcuts
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/cloud/manage-access/snowflake-external-oauth
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/core/connect-data-platform/snowflake-setup
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/dbt-ai/about-mcp
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/dbt-ai/setup-local-mcp
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/mesh/iceberg/apache-iceberg-support
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/docs/mesh/iceberg/databricks-iceberg-support
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/reference/resource-properties/data-formats
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/reference/resource-properties/unit-test-input
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/reference/resource-properties/unit-test-overrides
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/reference/resource-properties/unit-testing-versions
- https://docs-getdbt-com-git-key-pair-dbt-labs.vercel.app/reference/resource-properties/unit-tests

<!-- end-vercel-deployment-preview -->

runleonarun committed:
adding another note about Fusion an Snowflake key pairs

runleonarun commented on website/docs/docs/core/connect-data-platform/snowflake-setup.md:
```diff
@@ -107,6 +107,9 @@ To use key pair authentication, specify the `private_key_path` in your configura
+
 If you're using a 3DES-encrypted, headerless PEM body, you should either:
 - (Recommended) Re-export your existing key using a modern algorithm such as AES-256 encryption.
 - Add the `BEGIN` header and `END` footer to your PEM body. For example:
```
@amychen1776 should we be more opinionated and get rid of this second bullet?

amychen1776 replied:
I think it's okay to keep it but note that it's a workaround so it's clear that this is to unblock you, not what we recommend

amychen1776 replied:
I think we should also call out that if neither of those action are done - Fusion will error out.

runleonarun committed:
updating to make clear which is a workaround

runleonarun committed:
fixing error sentence

runleonarun commented:
> @runleonarun looks great! lol to the caps `DBT`

Ugh @amychen1776! Don't tell KJ, I will just take "dbt" out of the note title. That's the secodn time I've been burned by that all caps note title!! You'd think I'd learn.

runleonarun committed:
moving dbt from note title

matthewshaver commented:
@amychen1776 Is this a critical notice? We're about to split up the Core/Fusion installation pages and this note would be more fitting for that separate page. Do you think this could wait until the weeks end?

matthewshaver commented on website/docs/docs/core/connect-data-platform/snowflake-setup.md:
```diff
@@ -106,16 +106,26 @@ To use key pair authentication, specify the `private_key_path` in your configura
-If you're using a 3DES-encrypted, headerless PEM body, you should either:
-- (Recommended) Re-export your existing key using a modern algorithm such as AES-256 encryption.
-- Add the `BEGIN` header and `END` footer to your PEM body. For example:
+:::important
```
This doesn't really follow any established style for our callouts (bold header) and it's kind of long. What if we made this a H4 section instead?

matthewshaver replied:
```suggestion
```

matthewshaver commented on website/docs/docs/core/connect-data-platform/snowflake-setup.md:
````diff
@@ -106,16 +106,26 @@ To use key pair authentication, specify the `private_key_path` in your configura
+  < your existing encrypted private key contents >
   -----END ENCRYPTED PRIVATE KEY-----
   ```
 :::
````
See above feedback
```suggestion
```

matthewshaver commented on website/docs/docs/core/connect-data-platform/snowflake-setup.md:
```diff
@@ -106,16 +106,26 @@ To use key pair authentication, specify the `private_key_path` in your configura
-- (Recommended) Re-export your existing key using a modern algorithm such as AES-256 encryption.
-- Add the `BEGIN` header and `END` footer to your PEM body. For example:
+:::important
+**dbt Fusion engine requires modern key formats**
```
Moving modern down a level to keep the title short
```suggestion
#### dbt Fusion engine key formats
```

matthewshaver commented on website/docs/docs/core/connect-data-platform/snowflake-setup.md:
```diff
@@ -106,16 +106,26 @@ To use key pair authentication, specify the `private_key_path` in your configura
+:::important
+**dbt Fusion engine requires modern key formats**
+
+We recommend using PKCS#8 format with AES-256 encryption for key pair authentication with Fusion. Fusion's security requirements don't support legacy 3DES encryption or headerless key formats that work with dbt Core. Older key formats may cause authentication failures.
```
Let me know if you have questions about this suggestion
```suggestion
Fusion requires modern key formats and doesn't support legacy 3DES encryption or headerless keys.  We recommend using PKCS#8 format with AES-256 encryption for key pair authentication with Fusion. Using older key formats may cause authentication failures.
```

runleonarun committed:
Apply suggestions from code review

Co-authored-by: Matt Shaver <60105315+matthewshaver@users.noreply.github.com>

runleonarun commented on website/docs/docs/core/connect-data-platform/snowflake-setup.md:
````diff
@@ -106,19 +106,27 @@ To use key pair authentication, specify the `private_key_path` in your configura
+
+- (Temporary workaround) Add the `BEGIN` header and `END` footer to your PEM body:
 
   ```
````
```suggestion
  ```bash
```

runleonarun committed:
Update website/docs/docs/core/connect-data-platform/snowflake-setup.md

runleonarun committed:
move version block 

Reintroduced dbt Fusion engine key formats section with recommendations for key pair authentication and error handling.

runleonarun committed:
Clarify keypair config requirements for dbt Fusion

Added note about PKCS#8 format and AES-256 encryption requirements for dbt Fusion engine.
